### PR TITLE
add KodiVersion class with all comparison operators

### PR DESCRIPTION
**Pull request by Rechi**

### CODE REVIEW

File: kodi_addon_checker/KodiVersion.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    This file is part of Kodi - kodi.tv
+
+    SPDX-License-Identifier: GPL-3.0-only
+    See LICENSES/README.md for more information.
+"""
+
+from . import ValidKodiVersions
+
+class KodiVersion():
+    """docstring for KodiVersion"""
```
Comment (Razzeee) [inline kodi_addon_checker/KodiVersion.py]: Sees like a placeholder?

File: kodi_addon_checker/check_artwork.py [hunk 2]
```diff
@@ -13,21 +13,22 @@
...
 from .common import has_transparency, relative_path
+from .KodiVersion import KodiVersion
 from .record import INFORMATION, PROBLEM, WARNING, Record
 from .report import Report
 
 LOGGER = logging.getLogger(__name__)
 
 
-def check_artwork(report: Report, addon_path: str, parsed_xml, file_index: list, branch_name: str):
+def check_artwork(report: Report, addon_path: str, parsed_xml, file_index: list, kodi_version: str):
```
Comment (Razzeee) [inline kodi_addon_checker/check_artwork.py]: This is not a `str` right?

File: kodi_addon_checker/check_artwork.py [hunk 3]
```diff
@@ -39,7 +40,7 @@ def check_artwork(report: Report, addon_path: str, parsed_xml, file_index: list,
                     Record(PROBLEM, "Could not open image, is the file corrupted ? %s" % relative_path(image_path)))
 
 
-def _check_image_type(report: Report, image_type: str, parsed_xml, addon_path: str, branch_name: str):
+def _check_image_type(report: Report, image_type: str, parsed_xml, addon_path: str, kodi_version: str):
```
Comment (Razzeee) [inline kodi_addon_checker/check_artwork.py]: This is not a `str` right?

File: kodi_addon_checker/check_py3_compatibility.py [hunk 4]
```diff
@@ -44,7 +45,7 @@ def print_output(self, old_text, new_text, filename, equal):
         self.report.add(Record(self.log_level, relative_path(filename) + '\n' + diff[:-1]))
 
 
-def check_py3_compatibility(report: Report, path: str, branch_name: str):
+def check_py3_compatibility(report: Report, path: str, kodi_version: str):
```
Comment (Razzeee) [inline kodi_addon_checker/check_py3_compatibility.py]: This is not a `str` right?

File: kodi_addon_checker/schema_validation.py [hunk 5]
```diff
@@ -115,7 +116,7 @@ def _validate(xml, schemapath):
 
 
 def check_version(branch_name, schema_file):
-    all_branches = ['leia', 'krypton', 'jarvis', 'isengard', 'helix', 'gotham']
+    all_branches = ValidKodiVersions[::-1]
```
Comment (Razzeee) [inline kodi_addon_checker/schema_validation.py]: Why `::-1`?

Comment (Rechi) [reply kodi_addon_checker/schema_validation.py]: `ValidKodiVersions[::-1]` is `ValidKodiVersions` in reveres order

Comment (Razzeee) [reply kodi_addon_checker/schema_validation.py]: Thank you, only found :-1 when googling. Which said it would be the array minus the last item.

**Pull request merged**